### PR TITLE
Replace KServe with FastAPI for inference in finetuner example.

### PR DESCRIPTION
**Pull request by harubaru**
This PR replaces KServe's functionality with FastAPI for the sake of simplicity and allowing customers to get a better understanding of how to write an inference service on CoreWeave. Using KServe has a couple downsides, one of which is requiring package versions that conflicts with a key library being used such as [Tensorizer](https://github.com/coreweave/tensorizer).

This PR is to be added on top of #128.

### CODE REVIEW

**Comment by harubaru** [comment]
One thing that needs to be done so far is to share sampling code with `evaluate.py`, since this gets inference to _just_ work with FastAPI.

File: finetuner-workflow/finetuner/inference.py [hunk 1]
```diff
@@ -1,71 +1,77 @@
...
-    'TEMPERATURE': float(os.getenv('TEMPERATURE', 1.0)),
-    'TOP_K': int(os.getenv('TOP_K', 50)),
-    'TOP_P': float(os.getenv('TOP_P', 0.95)),
-    'REPETITION_PENALTY': float(os.getenv('REPETITION_PENALTY', 1.0)),
-    'NUM_RETURN_SEQUENCES': int(os.getenv('NUM_RETURN_SEQUENCES', 1)),
+args = {
+    "model": os.getenv("INFERENCE_MODEL", "distilgpt2"),
+    "device": int(os.getenv("INFERENCE_DEVICE", 0)),
+    "port": int(os.getenv("INFERENCE_PORT", 80))
 }
```
Comment (Eta0) [inline finetuner-workflow/finetuner/inference.py:41]: IMO using environment variables for this one while all of the other containers use command line arguments seems a little odd. Is there a particular reason to keep them this way, or just convenience?

**Review by Eta0 — CHANGES_REQUESTED** [review]
One required change (`async def` → `def`) and two nits. If you'd like me to whip up an `argparse` CLI for it I'd be happy to contribute one.

**Review by Eta0 — APPROVED** [review]
Looks good 👍

**Pull request merged**